### PR TITLE
[New Rule] Potential Active Directory Replication User Backdoor

### PR DESCRIPTION
**Pull request by w0rk3r**
## Issues

~~Blocked by https://github.com/elastic/integrations/issues/7381~~

Changes were merged and the PR was updated

## Summary

Identifies the modification of the nTSecurityDescriptor attribute in a domain object with rights related to DCSync to a user/computer account. Attackers can use this backdoor to re-obtain access to hashes of any user/computer.

### CODE REVIEW

**Review by Samirbous — APPROVED** [review]

**Review by terrancedejesus — APPROVED** [review]
Looks good. Can you run toml-lint command on it please before merging.

File: rules/windows/credential_access_dcsync_user_backdoor.toml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+[[rule.threat.technique]]
+id = "T1003"
+reference = "https://attack.mitre.org/techniques/T1003/"
+name = "OS Credential Dumping"
+
+    [[rule.threat.technique.subtechnique]]
+    id = "T1003.006"
+    reference = "https://attack.mitre.org/techniques/T1003/006/"
+    name = "DCSync"
```
Comment (Aegrah) [inline rules/windows/credential_access_dcsync_user_backdoor.toml:71]:
```suggestion
  [[rule.threat.technique]]
  id = "T1003"
  reference = "https://attack.mitre.org/techniques/T1003/"
  name = "OS Credential Dumping"

    [[rule.threat.technique.subtechnique]]
    id = "T1003.006"
    reference = "https://attack.mitre.org/techniques/T1003/006/"
    name = "DCSync"
```

File: rules/windows/credential_access_dcsync_user_backdoor.toml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+```
+"""
+references = [
+    "https://twitter.com/menasec1/status/1111556090137903104",
+    "https://www.specterops.io/assets/resources/an_ace_up_the_sleeve.pdf",
+    "https://github.com/SigmaHQ/sigma/blob/master/rules/windows/builtin/security/win_security_account_backdoor_dcsync_rights.yml"
+]
+risk_score = 73
+rule_id = "f8822053-a5d2-46db-8c96-d460b12c36ac"
+severity = "high"
```
Comment (Aegrah) [inline rules/windows/credential_access_dcsync_user_backdoor.toml:43]: Are we sure we should push this as high?

**Review by Aegrah — APPROVED** [review]
LGTM!

File: rules/windows/credential_access_dcsync_user_backdoor.toml (hunk 2 above)
Comment (w0rk3r) [reply rules/windows/credential_access_dcsync_user_backdoor.toml:43]: Good call, I will push it as medium for now

Comment (w0rk3r) [inline rules/windows/credential_access_dcsync_user_backdoor.toml:43]:
```suggestion
risk_score = 47
rule_id = "f8822053-a5d2-46db-8c96-d460b12c36ac"
severity = "medium"
```

File: rules/windows/credential_access_dcsync_user_backdoor.toml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+severity = "medium"
+tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Credential Access", "Data Source: Active Directory", "Use Case: Active Directory Monitoring"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.action:("Directory Service Changes" or "directory-service-object-modified") and event.code:"5136" and
+  winlog.event_data.AttributeLDAPDisplayName:"nTSecurityDescriptor" and
+  winlog.event_data.AttributeValue : (
+    (
```
Comment (Samirbous) [inline rules/windows/credential_access_dcsync_user_backdoor.toml:52]: can you add comment to explain those guids like DS-Replication-Get-Changes-All extended right

**Review by Samirbous — APPROVED** [review]

File: rules/windows/credential_access_dcsync_user_backdoor.toml (hunk 3 above)
Comment (w0rk3r) [reply rules/windows/credential_access_dcsync_user_backdoor.toml:52]: On the description?

**Review by terrancedejesus — APPROVED** [review]

File: rules/windows/credential_access_dcsync_user_backdoor.toml (hunk 3 above)
Comment (w0rk3r) [reply rules/windows/credential_access_dcsync_user_backdoor.toml:52]: As KQL doesn't support comments in the query, I'll add the rights documentation pages in the references

File: rules/windows/credential_access_dcsync_user_backdoor.toml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+Advanced Audit Policies Configuration >
+Audit Policies >
+DS Access >
+Audit Directory Service Changes (Success,Failure)
+```
+"""
+references = [
+    "https://twitter.com/menasec1/status/1111556090137903104",
+    "https://www.specterops.io/assets/resources/an_ace_up_the_sleeve.pdf",
+    "https://github.com/SigmaHQ/sigma/blob/master/rules/windows/builtin/security/win_security_account_backdoor_dcsync_rights.yml"
```
Comment (w0rk3r) [inline rules/windows/credential_access_dcsync_user_backdoor.toml:39]:
```suggestion
    "https://github.com/SigmaHQ/sigma/blob/master/rules/windows/builtin/security/win_security_account_backdoor_dcsync_rights.yml",
    "https://learn.microsoft.com/en-us/windows/win32/adschema/r-ds-replication-get-changes-all",
    "https://learn.microsoft.com/en-us/windows/win32/adschema/r-ds-replication-get-changes",
    "https://learn.microsoft.com/en-us/windows/win32/adschema/r-ds-replication-get-changes-in-filtered-set"
```

**Pull request merged**